test/index.spec.ts

```diff
@@ -137,452 +137,540 @@ test('encryptor:encrypt & decrypt with wrong password', async ({ page }) => {
+      const data = { foo: 'data to encrypt' };
+      const salt = await page.evaluate(() => window.encryptor.generateSalt());
+
+      const encryptedData = await page.evaluate(
+        async (args) => {
+          const key = await window.encryptor.keyFromPassword(
+            args.password,
+            args.salt,
+          );
+          return await window.encryptor.encryptWithKey(key, args.data);
+        },
+        { data, password, salt },
+      );
+      expect(Object.keys(encryptedData).sort()).toStrictEqual([
+        'data',
+        'iv',
+        'keyMetadata',
+      ]);
+
+      const encryptedString = JSON.stringify(
+        Object.assign({}, encryptedData, { salt }),
+      );
+      const decryptedData = await page.evaluate(
+        async (args) =>
+          await window.encryptor.decrypt(args.password, args.encryptedString),
+        { encryptedString, password },
       );
-      return await window.encryptor.encryptWithKey(key, args.data);
-    },
-    { data, password, salt },
-  );
-  expect(Object.keys(encryptedData).sort()).toStrictEqual(['data', 'iv']);
 
-  const encryptedString = JSON.stringify(
-    Object.assign({}, encryptedData, { salt }),
-  );
-  await expect(
-    page.evaluate(
-      async (args) =>
-        await window.encryptor.decrypt(
-          args.wrongPassword,
-          args.encryptedString,
+      expect(decryptedData).toStrictEqual(data);
+    });
+
+    test('encryptor:encrypt using key then decrypt using wrong password', async ({
+      page,
+    }) => {
+      const password = 'a sample passw0rd';
+      const wrongPassword = 'a wrong password';
+      const data = { foo: 'data to encrypt' };
+      const salt = await page.evaluate(() => window.encryptor.generateSalt());
+
+      const encryptedData = await page.evaluate(
+        async (args) => {
+          const key = await window.encryptor.keyFromPassword(
+            args.password,
+            args.salt,
+          );
+          return await window.encryptor.encryptWithKey(key, args.data);
+        },
+        { data, password, salt },
+      );
+      expect(Object.keys(encryptedData).sort()).toStrictEqual([
+        'data',
+        'iv',
+        'keyMetadata',
+      ]);
+
+      const encryptedString = JSON.stringify(
+        Object.assign({}, encryptedData, { salt }),
+      );
+      await expect(
+        page.evaluate(
+          async (args) =>
+            await window.encryptor.decrypt(
+              args.wrongPassword,
+              args.encryptedString,
+            ),
+          { encryptedString, wrongPassword },
         ),
-      { encryptedString, wrongPassword },
-    ),
-  ).rejects.toThrow('Incorrect password');
-});
+      ).rejects.toThrow('Incorrect password');
+    });
 
-test('encryptor:encrypt then decrypt using key', async ({ page }) => {
-  const password = 'a sample passw0rd';
-  const data = { foo: 'data to encrypt' };
+    test('encryptor:encrypt then decrypt using key', async ({ page }) => {
```

Comment:
This test doesn't rely on either `sampleEncryptedData` or `oldSampleEncryptedData`. Should it be inside the loop?